server/mailet/integration-testing/src/test/java/org/apache/james/mailets/SizeGreaterThanIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+            .sendMessageWithHeaders(BOUNCE_RECEIVER, RECIPIENT, "01234567\r\n".repeat(1025));
+
+        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(BOUNCE_RECEIVER, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessage(awaitAtMostOneMinute);
+    }
+
+    @Test
+    public void sizeGreaterThanMatcherShouldNotBounceWhenSizeWithinLimit() throws Exception {
+        jamesServer = TemporaryJamesServer.builder()
+            .withBase(MemoryJamesServerMain.SMTP_AND_IMAP_MODULE)
+            .withMailetContainer(
+                generateMailetContainerConfiguration())
+            .build(temporaryFolder.newFolder());
+
+        dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+        dataProbe.addUser(RECIPIENT, PASSWORD);
+        dataProbe.addUser(BOUNCE_RECEIVER, PASSWORD);
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .sendMessageWithHeaders(BOUNCE_RECEIVER, RECIPIENT, "01234567\r\n".repeat(1000));
+
+        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .awaitMessage(awaitAtMostOneMinute);
+    }
+
+    private MailetContainer.Builder generateMailetContainerConfiguration() {
+        return TemporaryJamesServer.DEFAULT_MAILET_CONTAINER_CONFIGURATION
+            .postmaster(POSTMASTER)
+            .putProcessor(transport())
+            .putProcessor(bounces());
+    }
+
+    private ProcessorConfiguration.Builder transport() {
+        // This processor delivers emails to BOUNCE_RECEIVER and POSTMASTER
+        // Other recipients will be bouncing
+        return ProcessorConfiguration.transport()
+            .addMailet(MailetConfiguration.BCC_STRIPPER)
+            .addMailet(MailetConfiguration.builder()
+                .matcher(RecipientIs.class)
+                .matcherCondition(BOUNCE_RECEIVER)
+                .mailet(LocalDelivery.class))
+            .addMailet(MailetConfiguration.builder()
+                .matcher(RecipientIs.class)
+                .matcherCondition(POSTMASTER)
+                .mailet(LocalDelivery.class))
+            .addMailet(MailetConfiguration.TO_BOUNCE);
+    }
+
+    public static ProcessorConfiguration.Builder bounces() {
+        return ProcessorConfiguration.bounces()
+            .addMailet(MailetConfiguration.builder()
+                .matcher(SizeGreaterThan.class)
+                .matcherCondition("10k")
+                .mailet(DSNBounce.class)
+                .addProperty("passThrough", "false")
+            )
+            .addMailet(MailetConfiguration.builder()
+                .matcher(All.class)
+                .mailet(LocalDelivery.class));
```

Comment:
You do local delivery in a bounce processor?

consider using a mail reposiotry instead (see ErrorTest)

Reply:
No really I find it really misleading to have more than one local delivery mailet, I find it misleading to have a localDelivery mailet within a bounce processor.

I am not expecting these kind of things from a mailet configuration.

Adressing this issue would help having a "real world oriented" scenario that would be easier to understand.